support.c

```diff
@@ -362,25 +385,26 @@ int parse_config_file(const char *file) {
     FILE *config_file;
     char line_buffer[256];
     int ctrl = 0;
+    char current_secret[256];
+    memset(current_secret, 0, sizeof(current_secret));
+
+    /* otherwise the list will grow with each call */
+    reset_config_variables();
 
     config_file = fopen(file, "r");
     if(config_file == NULL) {
         _pam_log(LOG_ERR, "Failed to open config file %s: %m", file);
         return 0;
     }
 
-    if (tac_source_addr != NULL) {
-        /* reset source address */
-        tac_source_addr = NULL;
-    }
-
-    char current_secret[256];
-    memset(current_secret, 0, sizeof(current_secret));

```

Comment:
I see you move 2 lines of code. However original location is better on performance.